applications/hybrid.md

```diff
@@ -0,0 +1,234 @@
+
+All events in all pallets that have identifying parameters will be indexed. For example the Transfer event in the Balances pallet is identifiable by the `AccountId` of both `from` and `to`.
+
+Other examples of identifying event parameters are `assetId` in the Assets pallet, `code_hash` in the contracts pallet, `CollectionId` and `ItemId` in the NFTs pallet, and `MultiLocation` in the XCM pallet.
+
+Reading blocks and writing indexes will occur in separate threads with a queue from the read thread to the write thread. Enabling reading and writing simultaneously will maximize indexing throughput.
+
+WSS queries will be handled via the highly scalable [tokio_tungstenite](https://github.com/snapview/tokio-tungstenite) Rust library.
+
+In addition to the identifier being searched for, queries will be able to include start block, offset, and limit to control which events are returned.
+
+Consumers will be able to subscribe for new events that match a query. This will be handles by sled's powerful [watch_prefix](https://docs.rs/sled/latest/sled/struct.Tree.html#method.watch_prefix) functionality.
+
+The database keys will be constructed in such a way so that events can be found using iterators starting at a specific block number. For example, for for the AccountId keyspace:
+
+`AccountId/BlockNumber/EventIndex`
+
+
+#### Dapp
+
+The Hybrid dapp will be a [Vue](https://vuejs.org/) dapp, using the [Vuetify](https://vuetifyjs.com) framework for the user interface. [pnpm](https://pnpm.io/) and [Vite](https://vitejs.dev/) will be used for the build.
+
+It will use [@polkadot/api](https://github.com/polkadot-js/api) to retrieve data from the chain, either via the Substrate Connect light client, or via an RPC connection to a full archival node. The Hybrid indexer will be queried via WSS.
+
+This grant will only include a basic dapp including:
+- block browsing, showing transactions and events
+- event searching, e.g. searching for all events connected to a specific `AccountId`.
+
+A subsequent grant application will be made to develop a richer dapp experience.
+
+
+### Ecosystem Fit
+
+A major issue holding back the Substrate and Polkadot ecosystem is a high quality block explorer comparable to Etherscan. We need such a block explorer to be:
+- non-proprietary
+- as decentralized as possible
+- not requiring massive system resources
+- feature-rich
+
+The target audience is blockchain enthusiasts and developers. Eventually end-users should not need to know about block explorers, but this depends on dapps improving their user experience.
+
+The indexing component has value far beyond the Hybrid block explorer. Many Substrate applications will find a centralized event indexer extremely useful.
+
+#### Comparison to other Substrate explorers / indexers
```

Comment:
Do you know of any examples of _decentralised_ explorers? Perhaps in other ecosystems?

Reply:
I actually funded a decentralized Ethereum explorer years ago:
https://github.com/acuity-social/block-explorer

It is similar to polkadot.js Apps, in that there is no indexing.

LibreScan was announced a year ago, but doesn't seem to have gone anywhere.

I don't of anyone either building a real decentralized indexing block explorer, or a hybrid one like in this application.